### PR TITLE
Fix silly npm warnings

### LINKED ISSUE #10251: Warnings in SignalR build due to invalid peer dependencies
```
F:\workspace\_work\1\s\eng\targets\Npm.Common.targets(21,5): warning : " > ts-jest@22.4.6" has incorrect peer dependency "jest@^22.4.0 || ^22.5.0-alpha.1 || ^23.0.0-alpha.1". [F:\workspace\_work\1\s\src\SignalR\clients\ts\common\common.npmproj]
F:\workspace\_work\1\s\eng\targets\Npm.Common.targets(21,5): warning : " > karma-summary-reporter@1.5.0" has incorrect peer dependency "karma@^1.2.0 || 2.x". [F:\workspace\_work\1\s\src\SignalR\clients\ts\FunctionalTests\SignalR.Npm.FunctionalTests.npmproj]
```

Build: https://dev.azure.com/dnceng/internal/_build/results?buildId=189015

We are seeing these warnings in our CI and in CTI testing (to validate that the latest VS can build our code): https://devdiv.visualstudio.com/DevDiv/_workitems/edit/892389?src=WorkItemMention&src-action=artifact_link

We should see if we can resolve these warnings since Warnings Are Bad:tm:

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/aspnet/AspNetCore/issues/10251

@anurse Who does dependency update things now? (This is test only)

### CODE REVIEW

**Comment by anurse** [comment]
> @anurse Who does dependency update things now? (This is test only)

The formal "compliance" stuff is automated in our build through the Component Governance tools. I think @Pilchie should be notified though, just as @Eilon was in the past despite the automatic stuff.

In this case, the dependency is being updated in order to fix NPM warnings during installation (see linked issue) and is used only in test code. I approve ✔️ of this update.

**Review by analogrelay — APPROVED** [review]
Looks like the `ts-jest` update barfed on the lock file but c'est la vie.

**Review by halter73 — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Also looks like I forgot to run tests locally, and the update broke some stuff

**Comment by Eilon** [comment]
Yup, please let @Pilchie know about significant OSS updates. For me the most important ones were ones that affect the shipping product (even minor updates). For test/build-only assets I mostly cared about significant changes (e.g. new dependency).

File: src/SignalR/clients/ts/FunctionalTests/tsconfig.jest.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/tsconfig.jest.json]: `ts-jest` changed something with how they compiled files which made the types now actually needed, so had to add a new config file to have the correct `typeRoots` and use that for FunctionalTests

**Comment by BrennanConroy** [comment]
:up: 📅

File: src/SignalR/clients/ts/FunctionalTests/tsconfig.jest.json (hunk 1 above)
Comment (halter73) [reply src/SignalR/clients/ts/FunctionalTests/tsconfig.jest.json]: 👍

**Comment by halter73** [comment]
Still approved.

**Pull request merged**